presto-main/src/main/java/io/prestosql/operator/scalar/timestamp/HumanReadableSeconds.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar.timestamp;
+
+import io.airlift.slice.Slice;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.type.StandardTypes;
+
+import java.util.StringJoiner;
+import java.util.concurrent.TimeUnit;
+
+import static io.airlift.slice.Slices.utf8Slice;
+
+@ScalarFunction("human_readable_seconds")
+public final class HumanReadableSeconds
+{
+    private static final String HUMAN_REDABLE_FORMAT = "%s, %s, %s, %s, %s";
+    private static final int DAYS_IN_WEEK = 7;
+    private static final String WEEK = "week";
+    private static final String DAY = "day";
+    private static final String HOUR = "hour";
+    private static final String MINUTE = "minute";
+    private static final String SECOND = "second";
+
+    private HumanReadableSeconds() {}
+
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice humanReadable(@SqlType(StandardTypes.DOUBLE) double inputSeconds)
+    {
+        long seconds = Double.valueOf(inputSeconds).longValue();
+
+        long weeks = processWeeks(seconds);
+        if (weeks > 0) {
+            seconds -= TimeUnit.DAYS.toSeconds(weeks * DAYS_IN_WEEK);
+        }
+
+        long days = processDays(seconds);
+        seconds -= TimeUnit.DAYS.toSeconds(days);
+
+        long hours = processHours(seconds);
+        seconds -= TimeUnit.HOURS.toSeconds(hours);
+
+        long minutes = processMinutes(seconds);
+        seconds -= TimeUnit.MINUTES.toSeconds(minutes);
+
+        long lseconds = processSeconds(seconds);
+
+        return getTimePeriods(weeks, days, hours, minutes, lseconds);
+    }
+
+    private static long processWeeks(@SqlType(StandardTypes.BIGINT) long seconds)
```

Comment:
`@SqlType(StandardTypes.BIGINT)` is not needed for these methods. The annotation is used by Presto to discover metadata about functions, but these are implementation details internal to the function